src/rpc/exceptions.hpp

```diff
@@ -19,19 +19,30 @@ namespace memgraph::rpc {
 /// `utils::BasicException` is used for transient errors that should be reported
 /// to the user and `utils::StacktraceException` is used for fatal errors.
 /// This exception always requires explicit handling.
-class RpcFailedException final : public utils::BasicException {
+class RpcFailedException : public utils::BasicException {
  public:
-  RpcFailedException(const io::network::Endpoint &endpoint)
-      : utils::BasicException::BasicException(
-            "Couldn't communicate with the cluster! Please contact your "
-            "database administrator."),
-        endpoint_(endpoint) {}
+  RpcFailedException(std::string_view msg) : utils::BasicException(msg) {}
+  SPECIALIZE_GET_EXCEPTION_NAME(RpcFailedException);
+};
 
-  /// Returns the endpoint associated with the error.
-  const io::network::Endpoint &endpoint() const { return endpoint_; }
-  SPECIALIZE_GET_EXCEPTION_NAME(RpcFailedException)
+class VersionMismatchRpcFailedException : public RpcFailedException {
```

Comment:
Don't we want to add `final` to this classes? Not sure if that was a convention or not.

Reply:
IMO minimal benefit in this situation, not going to add it.